src/Native/Runtime/gcrhenv.cpp

```diff
@@ -786,6 +787,10 @@ COOP_PINVOKE_HELPER(void, RhpBox, (Object * pObj, void * pData))
     {
         memcpy(pbFields, pData, cbFields);
     }
+#else
+    //TODO: write barriers currently crash in wasm and we dont really have GC anyway
```

Comment:
```
__Z6AssertPKcS0_jS0_ (wasm-function[2750]:42)
at __Z32GCSafeCopyMemoryWithWriteBarrierPvPKvj (wasm-function[4857]:128)
at _RhpBox (wasm-function[4884]:218)
at _S_P_CoreLib_System_Runtime_RuntimeExports__RhBox (wasm-function[184]:859)
at _S_P_CoreLib_System_Runtime_RuntimeExports__RhBoxAny (wasm-function[183]:546)
at _StartupCodeMain (wasm-function[389]:3904)
```

digging into the inlines that make up GCSafeCopyMemoryWithWriteBarrier There are a lot of asserts that could be triggered inside InlineForwardGCSafeCopy, but they all appear to be alignment related. I'll see if there is something I can enforce around the structure being boxed.

Reply:
I've filed #5176 to align the shadow stack since it periodically causes annoyances like this. You don't need to deal with it in this change if you can find a way to avoid the crash though.

Reply:
Now that the alignment change is merged, can you undo this?

Reply:
I missed this in my last commit, I'll make a quick PR to put it back